Standards/scs-0214-v1-k8s-node-distribution.md

```diff
@@ -0,0 +1,91 @@
+| Machine | Virtual or bare-metal entity with computational capabilities |
+
+## Motivation
+
+In normal day-to-day operation, it is not unusual for some operational failures, either
+due to wear and tear of hardware, software misconfigurations, external problems or
+user errors. Whichever was the source of such an outage, it always means down-time for
+operations and users and possible even data loss.
+Therefore, a Kubernetes cluster in a productive environment should be distributed over
+multiple "failure zones" in order to provide fault-tolerance and high availability.
+This is especially important for the control plane of the cluster, since it contains the
+state of the whole cluster. A failure of this component could mean an unrecoverable failure
+of the whole cluster.
+
+## Design Considerations
+
+Most design considerations of this standard follow the previously written Decision Record
+"Kubernetes Nodes Anti Affinity" as well as the Kubernetes documents about [High Availability]
+and [Best practices for large clusters].
+
+SCS wishes to prefer distributed, highly-available systems due to their obvious advantages
+like fault-tolerance and data redundancy. But it also understands the costs and overhead
+for the providers associated with this effort, since the infrastructure needs to have
+hardware which will just be used to provide fail-over safety or duplication.
+
+The document [Best practices for large clusters] describes the concept of a failure zone.
+This term isn't defined any further, but can in this context be described as a number of
+physical (computing) machines in such a vicinity to each other (either through physical
+or logical interconnection in some way), that specific problems inside this zone would put
+all these machines at risk of failure/shutdown. It is therefore necessary for important
+data or services to not be present just on one failure zone.
+How such a failure zone should be defined is dependent on the risk model of the service/data
+and its owner as well as the capabilities of the provider. Zones could be set from things
+like single machines or racks up to whole datacenters or even regions, which could be
+coupled by things like electrical grids. They're therefore purely logical entities, which
+shouldn't be defined further in this document.
+
+## Decision
+
+This standard formulates the requirement for the distribution of Kubernetes nodes in order
+to provide a fault-tolerant and available Kubernetes cluster infrastructure.
+Since some providers only have small environments to work with and therefore couldn't
+comply with this standard, it will be treated as a RECOMMENDED standard, where providers
+can OPT-OUT.
+
+If the standard is used by a provider, the following decisions are binding and valid:
+
+- The control plane nodes SHOULD be distributed over multiple physical machines. Kubernetes
```

Comment:
In the section above the term "failure zone" is used multiple times. I personally have the feeling that "multiple physical machines" does not fit the intention perfectly. We would like that machines (as defined in the glossary, so we mean nodes here) should be placed in different failure zones to not being affected by issues on routers, burning facilities etc. as mentioned below this comment. That means I would remove the word "physical".

Reply:
IMO, I would leave the part about `physical machines` in. Even if the concept of a `failure zone` is the important term here to achieve availability security, it's nonetheless dependent on physical separation underneath (since a machine could always fail for some reason). If this wouldn't be defined, a failure zone could just be a few VMs and the next failure zone could be another set of VMs on the same machine. This isn't how failure zones should be used, but it could happen nonetheless.
That's why I would keep the physical seperation in there